### DISCUSSION TITLE
Swift Realtime presence

### DISCUSSION THREAD

**Discussion by miguel-arrf** [supabase/supabase#12968] [Questions]
Hi!

I've been trying to follow this guide, [https://supabase.com/docs/guides/realtime/presence](https://supabase.com/docs/guides/realtime/presence) to create a Swift realtime presence feature on my app.

```
 var channel = realTimeClient.channel(.schema("public"), params: ["presence":"user1"])
        
        channel.onMessage { message in
            print("message: \(message.payload)")
            return message
        }
        channel.push(.join, payload: ["presence":"user2"])
        channel.subscribe()
```

For example, with something like this, I receive the following error:
`Realtime/Channel.swift:466: Fatal error: Tried to push join to schema("public") before joining. Use channel.join() before pushing events``
However, the channel has no public `.join` method.

Furthermore, the endpoint being used for realtime is `v1`. Are there any plans to update it to `v2`?

**Comment by DannieMcnortongYo** [comment]
It looks like the error message you are receiving is indicating that you are trying to push a message before joining the channel. Make sure to call channel.join() before pushing any messages to the channel.

Regarding the Supabase Realtime endpoint, it appears that v1 is currently the latest version. However, Supabase is constantly updating and improving their platform, so it's possible that a new version of the Realtime endpoint will be released in the future.

**Reply by miguel-arrf** [reply]
Thank you for the reply!

I should have been a little bit more explicit, I'm sorry. There is no `.join()` method. At least a public one. 

😕

**Comment by w3b6x9** [comment]
@miguel-arrf did you roll your own Realtime Swift client or using a community one? If it's the latter can you share the client lib?

> Are there any plans to update it to v2?

we have no plans at the moment.

**Reply by miguel-arrf** [reply]
Hi!

Actually, I've created a wrapper for the JS Realtime client, and I'm using it in a WebView that is initialized in the background while the application (SwiftUI) is starting. It is embedded in a ObservableObject so that I can receive updates via @Published variables.

I'll try to organize the code and share it. It may be usefull for some people :)
